### ISSUE TITLE
Show subcategories

### ISSUE THREAD

**Issue by khatchad** [google/shipshape#115]
Please add an option to show subcategories of the various categories displayed upon using the --categories command line option. Perhaps --subcategories can take a category argument, e.g., :
```
shipshape --subcategories error-prone
```

**Comment by ciera** [comment]
This is going to be somewhat difficult. Some of the analysis tools do not make the subcategories easy to determine. There are others that create the subcategory dynamically.

**Comment by supertriceratops** [comment]
We could add a "GetSubcategories" call to the individual analyzer API...

**Comment by ciera** [comment]
Yes, the problem is how the analyzers actually implement that. Most of the open source analyzers do not make getting this easy. Additionally, it implies that there *is* a static list. It prevents analyzers from being able to dynamically generate subcategory names.
